docs/usage/openapi-definition-resolver.md

```diff
@@ -51,6 +51,36 @@ SwaggerClient.resolve({ url: 'https://raw.githubusercontent.com/swagger-api/swag
 > *__Note:__ you can provide `Swagger.resolve` both `spec` and `url` options. In that case
 `spec` will always be preferred and `url` option will be ignored.* 
 
+#### Authentication
+
+When JSON-References points to a remote document protected by authentication,
+`requestInterceptor` option can be used to provide e.g. an authentication header.
+
+```js
+import SwaggerClient from 'swagger-client';
+
+const requestInterceptor = request => {
+  if (request.url.startsWith('https://raw.githubusercontent.com/swagger-api/')) {
```

Comment:
@char0n Would it be better to use `if (request.loadSpec)` here? Or at least mention `loadSpec` somewhere in this section. If I remember correctly, `loadSpec: true` is set any time Swagger JS fetches a document - this includes both the main definition specified by `url` and any external `$refs`.

Reply:
@hkosova right I've overlooked  `loadSpec` property when consolidating new documentation. Checkout the PR now. I've added a full new section related to `loadSpec` and couple of examples. Check it out and let me know If I could change/add something more to make it more understandable to SwaggerClient consumers.